src/lib/classes/QTheme.svelte.ts

```diff
@@ -1,71 +1,32 @@
-import materialDynamicColors from "material-dynamic-colors";
-import QColors from "$lib/utils/colors";
-import type {
-  IMaterialDynamicColorsTheme,
-  IMaterialDynamicColorsThemeColor,
-} from "material-dynamic-colors/src/cdn/interfaces";
-import { convertCase } from "$lib/utils/string";
+import { generateColors } from "$lib/utils/colors";
+import type { HexValue, QuaffColors } from "$lib/utils/colors";
 
-interface MaterialColors {
-  primary: string;
-  "on-primary": string;
-  "primary-container": string;
-  "on-primary-container": string;
-  secondary: string;
-  "on-secondary": string;
-  "secondary-container": string;
-  "on-secondary-container": string;
-  tertiary: string;
-  "on-tertiary": string;
-  "tertiary-container": string;
-  "on-tertiary-container": string;
-  error: string;
-  "error-container": string;
-  "on-error": string;
-  "on-error-container": string;
-  background: string;
-  "on-background": string;
-  surface: string;
-  "on-surface": string;
-  "surface-variant": string;
-  "on-surface-variant": string;
-  outline: string;
-  "inverse-on-surface": string;
-  "inverse-surface": string;
-  "inverse-primary": string;
-  shadow: string;
-}
-
-interface MaterialColorsTheme {
-  light: MaterialColors;
-  dark: MaterialColors;
-}
-
-type ThemeColorName = `${keyof MaterialColors}-${keyof MaterialColorsTheme}`;
-type ThemeColors = Record<ThemeColorName, string>;
+type Mode = "light" | "dark";
+type ThemeColors = Record<`${keyof QuaffColors}-${Mode}`, HexValue>;
 
 function extractColorFromCssVar(cssVar: string) {
   const rootStyles = getComputedStyle(document.documentElement);
-  const varName = cssVar.replace(/var\(([a-z0-9-]+)\)/, "$1");
+  const varName = cssVar.replace(/var\(([a-z0-9-]+)\)/, "$1"); // remove var(...) to get only the color
```

Comment:
If possible, comments should be on top, not at the end of the line.

Reply:
Ok

Reply:
06f89276391b28572232d73311d599b3ac986aa3